lib/IdApi.php

```diff
@@ -0,0 +1,76 @@
+<?php
+spl_autoload_register(function($class) {
+    require_once($class.'.php');
+});
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Http\Message\ResponseInterface;
+
+require 'vendor/autoload.php';
+
+class IdApi
+{
+    const SID_SERVERS = [
+        'https://testapi.smileidentity.com/v1',
+        'https://api.smileidentity.com/v1'
+    ];
+    public Signature $sig_class;
+    private String $partner_id;
+    private String $default_callback;
+    private String $sid_server;
+
+    /**
+     * IdApi constructor.
+     * @param $partner_id
+     * @param $default_callback
+     * @param $api_key,
+     * @param $sid_server
+     * @throws Exception
+     */
+    public function __construct($partner_id, $default_callback, $api_key, $sid_server)
+    {
+        $this->partner_id = $partner_id;
+        $this->default_callback = $default_callback;
+        $this->sig_class = new Signature($api_key, $partner_id);
+        if(strlen($sid_server) == 1) {
+            if(intval($sid_server) < 2) {
+                $this->sid_server = self::SID_SERVERS[intval($sid_server)];
+            } else {
+                throw new Exception("Invalid server selected");
+            }
+        } else {
+            $this->sid_server = $sid_server;
+        }
+    }
+
+    /**
+     * @param $partner_params
+     * @param $id_info
+     * @param $use_async
+     * @return ResponseInterface
+     * @throws GuzzleException
+     */
+    public function submit_job($partner_params, $id_info, $use_async, $guzzle = null): ResponseInterface
+    {
+        $b = $this->sig_class->generate_sec_key();
+        $sec_key = $b[0];
+        $timestamp = $b[1];
+
+        $data = array(
+            'language' => 'php',
+            'callback_url' => $this->default_callback,
```

Comment:
There are no validations on the inputs in this class. Refer to our other libraries such as JS or ruby for the validations that we perform.